### PR TITLE
Lists of types in preferences API

### PR DESCRIPTION
**Pull request by andydotxyz**
### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: app/preferences.go [hunk 1]
```diff
@@ -109,6 +109,10 @@ func (p *preferences) loadFromFile(path string) (err error) {
 
 	p.InMemoryPreferences.WriteValues(func(values map[string]interface{}) {
 		err = decode.Decode(&values)
+		if err != nil {
+			return
```
Comment (Jacalz) [inline app/preferences.go:113]: Should we not do a `fyne.LogError()` on the value?

**Review by Jacalz — COMMENTED** [review]
Looks good. Seems like a nice addition. Just one comment inline.

File: app/preferences.go (hunk 1 above)
Comment (andydotxyz) [reply app/preferences.go:113]: I don't think so, no - this is exiting the write before applying conversion with an error set to the return value on the parent function call. So the caller will get the error and should log it if required.

Comment (Jacalz) [reply app/preferences.go:113]: Alright. Sounds good.

**Review by Jacalz — APPROVED** [review]

**Pull request merged**